### PR TITLE
Add JDK8/JDK17 into CI matrix

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR adds JDK8/JDK17 into the CI build matrix. Remember to also update the github branch status checks after the PR gets merged.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,21 +23,35 @@ jobs:
...
         with:
           fetch-depth: 0
 
+      - name: Setup Java (temurin@8)
+        if: matrix.java == 'temurin@8'
+        uses: actions/setup-java@v2
+        with:
+          distribution: temurin
+          java-version: 8
+
```
Comment (johanandren) [inline .github/workflows/ci.yml]: Given that Java 8 was eol 2019 I think we can skip that, adding 17 is good though.

```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,21 +23,35 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         scala: [2.12.15, 2.11.12, 2.13.8]
-        java: [temurin@11]
+        java: [temurin@8, temurin@11, temurin@17]
```
Comment (johanandren) [inline .github/workflows/ci.yml]:
```suggestion
        java: [temurin@11, temurin@17]
```

File: build.sbt [hunk 3]
```diff
@@ -68,6 +68,12 @@ developers := List(
   Developer("johanandren", "Johan Andrén", "johan@markatta.com", url("https://markatta.com/johan/codemonkey"))
 )
 
+ThisBuild / githubWorkflowJavaVersions := List(
+  JavaSpec.temurin("8"),
```
Comment (johanandren) [inline build.sbt]:
```suggestion
```

**Comment by mdedetrich** [comment]
@johanandren  Just updated + rebased the PR with JDK 8 removed

**Review by johanandren — APPROVED** [review]

**Pull request merged**